pymeasure/instruments/keithley/keithley2450.py

```diff
@@ -0,0 +1,653 @@
+    wires = Instrument.control(
+        ":SENS:RES:RSENSE?", ":SENS:RES:RSENSE %d",
+        """ An integer property that controls the number of wires in
+        use for resistance measurements, which can take the value of
+        2 or 4.
+        """,
+        validator=strict_discrete_set,
+        values={4:1, 2:0},
+        map_values=True
+    )
+    # TODO: test on instrument
+    buffer_points = Instrument.control(
+        ":TRAC:POIN?", ":TRAC:POIN %d",
+        """ An integer property that controls the number of buffer points. This
+        does not represent actual points in the buffer, but the configuration
+        value instead. """,
+        validator=truncated_range,
+        values=[1, 6875000],
+        cast=int
+    )
+    # TODO: test on instrument
+    means = Instrument.measurement(
+        ":TRACe:STATistics:AVERage?",
+        """ Reads the calculated means (averages) for voltage,
+        current, and resistance from the buffer data  as a list. """
+    )
+    # TODO: test on instrument
+    maximums = Instrument.measurement(
+        ":TRACe:STATistics:MAXimum?",
+        """ Returns the calculated maximums for voltage, current, and
+        resistance from the buffer data as a list. """
+    )
+    # TODO: test on instrument
+    minimums = Instrument.measurement(
+        ":TRACe:STATistics:MINimum?",
+        """ Returns the calculated minimums for voltage, current, and
+        resistance from the buffer data as a list. """
+    )
+    # TODO: test on instrument
+    standard_devs = Instrument.measurement(
+        ":TRACe:STATistics:STDDev?",
+        """ Returns the calculated standard deviations for voltage,
+        current, and resistance from the buffer data as a list. """
+    )
+
+    ###########
+    # Trigger #
+    ###########
+
+    # TODO: Trigger system will need a complete reimplementation because the model has changed significantly from 2400    
+
+    def __init__(self, adapter, **kwargs):
+        super(Keithley2450, self).__init__(
+            adapter, "Keithley 2450 SourceMeter", **kwargs
+        )
+
+    # FIXME: Ok
+    def enable_source(self):
+        """ Enables the source of current or voltage depending on the
+        configuration of the instrument. """
+        self.write("OUTPUT ON")
+
+    # FIXME: Ok
+    def disable_source(self):
+        """ Disables the source of current or voltage depending on the
+        configuration of the instrument. """
+        self.write("OUTPUT OFF")
+
+    # FIXME: Fixed
+    def measure_resistance(self, nplc=1, resistance=2.1e5, auto_range=True):
+        """ Configures the measurement of resistance.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param resistance: Upper limit of resistance in Ohms, from -210 MOhms to 210 MOhms
+        :param auto_range: Enables auto_range if True, else uses the set resistance
+        """
+        log.info("%s is measuring resistance." % self.name)
```

Comment:
`log.info` messages should be avoided. You could drop them to `log.debug`, but I think it is preferable not to have them at all, because it doesn't give the library user a choice.

Reply:
Can you elaborate on that point about choice? Info-Level Messages can be routed and enabled/disabled like any other, and users can selectively set the level of the logger in just this module to adjust what level they want, no?

Reply:
I used the Keithley2400 object as the basis for this pull request which has the `log.info` messages. Overall I am with @bilderbuchi in that these info messages don't show up by default since there is no log handler created by the code. If a user specifies a log handler it's probably because they want the added verbosity.
I am happy to remove the messages if that would bring the code in more consistency with the rest of the package though.

Reply:
Those are fair points @bilderbuchi. @hlgirard, let's keep it as is.